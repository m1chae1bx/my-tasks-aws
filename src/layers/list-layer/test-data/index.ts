import { ListDetails } from "../list.model";

export const testList: ListDetails = {
  id: "test-list-id",
  name: "Test list 1",
  userId: "test-user-id",
  isDefault: true,
};

export const testLists: ListDetails[] = [
  {
    id: "test-list-id-1",
    name: "Test list 1",
    userId: "test-user-id",
    isDefault: true,
  },
  {
    id: "test-list-id-2",
    name: "Test list 2",
    userId: "test-user-id",
    isDefault: false,
  },
];
