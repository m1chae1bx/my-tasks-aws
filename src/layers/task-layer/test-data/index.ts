import { DueDate, GetTasksQuery, TaskDetails } from "../task.model";

export const testTask: TaskDetails = {
  id: "test-task-id",
  listId: "test-list-id",
  name: "Test task",
  isCompleted: false,
  dueDate: new Date("2022-01-01"),
  desc: "Test task description",
};

export const completedTask: TaskDetails = {
  ...testTask,
  isCompleted: true,
};

export const testTasks: Partial<TaskDetails>[] = [
  {
    id: "test-task-id-1",
    name: "Test task 1",
    desc: "Test task description",
    isCompleted: false,
    dueDate: new Date("2020-01-01T00:00:00.000Z"),
    listId: "test-list-id",
  },
  {
    id: "test-task-id-2",
    name: "Test task 2",
    desc: "Test task description",
    isCompleted: true,
    dueDate: new Date("2020-01-01T00:00:00.000Z"),
    listId: "test-list-id",
  },
];

export const getTasksQuery: GetTasksQuery = {
  name: "Test task",
  dueDate: DueDate.TODAY,
  today: "2020-01-01",
  includeCompleted: "true",
};
