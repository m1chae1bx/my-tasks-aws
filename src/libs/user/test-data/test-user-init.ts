import { UserDetails, UserPreferences } from "../user.model";

const userPreferences: UserPreferences = {
  defaultListId: "test-default-list-id",
};

export const testUserInit: UserDetails = {
  id: "testUserId",
  email: "test@email.com",
  fullName: "Full Name",
  nickname: "Nickname",
  hash: "test-hash",
  salt: "test-salt",
  preferences: userPreferences,
};
