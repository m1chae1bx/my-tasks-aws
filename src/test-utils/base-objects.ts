import {
  APIGatewayProxyEvent,
  APIGatewayTokenAuthorizerEvent,
} from "aws-lambda";
import { AWSError } from "aws-sdk";
import { DocumentClient } from "aws-sdk/lib/dynamodb/document_client";
import { PromiseResult } from "aws-sdk/lib/request";

export const baseAPIGatewayProxyEvent: APIGatewayProxyEvent = {
  headers: {},
  multiValueHeaders: {},
  httpMethod: "",
  isBase64Encoded: false,
  path: "",
  pathParameters: null,
  queryStringParameters: null,
  multiValueQueryStringParameters: null,
  stageVariables: null,
  requestContext: {
    accountId: "",
    apiId: "",
    authorizer: {},
    protocol: "",
    httpMethod: "",
    path: "",
    stage: "",
    requestId: "",
    resourceId: "",
    resourcePath: "",
    requestTimeEpoch: 0,
    identity: {
      accessKey: "",
      accountId: "",
      apiKey: "",
      apiKeyId: "",
      caller: "",
      clientCert: null,
      cognitoAuthenticationProvider: null,
      cognitoAuthenticationType: null,
      cognitoIdentityId: null,
      cognitoIdentityPoolId: null,
      principalOrgId: null,
      sourceIp: "",
      user: null,
      userAgent: null,
      userArn: null,
    },
  },
  body: null,
  resource: "",
};

export const baseTokenAuthorizerEvent: APIGatewayTokenAuthorizerEvent = {
  authorizationToken: "Bearer test-bearer-token",
  methodArn: "test-method-arn",
  type: "TOKEN",
};

export const ajvError = [
  {
    keyword: "",
    instancePath: "",
    schemaPath: "",
    params: {},
    message: "test error",
  },
];

export const ddbDeleteResult: PromiseResult<
  DocumentClient.DeleteItemOutput,
  AWSError
> = {
  $response: {
    data: {},
    error: undefined,
    hasNextPage: jest.fn(),
    httpResponse: {
      statusCode: 200,
      headers: {},
      statusMessage: "OK",
      body: JSON.stringify({}),
      createUnbufferedStream: jest.fn(),
      streaming: false,
    },
    nextPage: jest.fn(),
    redirectCount: 0,
    requestId: "test-request-id",
    retryCount: 0,
  },
};

export const conditionalCheckFailedException: AWSError = {
  code: "ConditionalCheckFailedException",
  name: "ConditionalCheckFailedException",
  message: "The conditional request failed",
  time: new Date("2022-01-01T00:00:00.000Z"),
};
