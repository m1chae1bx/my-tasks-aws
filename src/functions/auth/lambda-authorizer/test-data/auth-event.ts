import { baseTokenAuthorizerEvent } from "@test-utils/base-objects";

export const authEvent = baseTokenAuthorizerEvent;
