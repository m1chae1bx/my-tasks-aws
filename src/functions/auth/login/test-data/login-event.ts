import { baseAPIGatewayProxyEvent } from "@test-utils/base-objects";

export const loginEvent = {
  ...baseAPIGatewayProxyEvent,
  body: JSON.stringify({
    id: "testUserId",
    password: "secret-password",
  }),
};
