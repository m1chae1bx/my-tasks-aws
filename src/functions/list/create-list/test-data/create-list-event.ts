import { baseAPIGatewayProxyEvent } from "@test-utils/base-objects";

export const createListEvent = {
  ...baseAPIGatewayProxyEvent,
  pathParameters: {
    userId: "testUserId",
  },
  body: JSON.stringify({
    name: "testListName",
    isDefault: false,
  }),
};
