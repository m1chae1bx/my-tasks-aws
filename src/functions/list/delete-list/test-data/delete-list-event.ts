import { baseAPIGatewayProxyEvent } from "@test-utils/base-objects";

export const deleteListEvent = {
  ...baseAPIGatewayProxyEvent,
  pathParameters: {
    userId: "testUserId",
    listId: "testListId",
  },
};
