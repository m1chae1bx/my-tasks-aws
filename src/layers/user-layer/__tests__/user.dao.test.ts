import { conditionalCheckFailedException } from "@test-utils/base-objects";
import { testUserInit } from "../test-data/test-user-init";
import {
  EnvironmentConfigError,
  ErrorCode,
  RequiredPropertyMissingError,
  UserNotFoundError,
} from "/opt/nodejs/errors";

describe("create", () => {
  describe("happy path", () => {
    it("should return the user ID", async () => {
      jest.resetModules();
      jest.doMock("/opt/nodejs/dynamo.config", () => ({
        dynamoClient: {
          put: jest.fn().mockReturnValueOnce({
            promise: jest.fn().mockResolvedValueOnce(null),
          }),
        },
        TABLE_NAME: "test-table-name",
      }));
      const userDao = await import("../user.dao");
      const result = await userDao.create(testUserInit);
      expect(result).toBe(testUserInit.id);
    });
  });

  describe("sad path - DynamoDB table undefined", () => {
    it("should throw a EnvironmentConfigError error", async () => {
      let err: unknown;

      jest.resetModules();
      jest.doMock("/opt/nodejs/dynamo.config", () => ({
        TABLE_NAME: undefined,
      }));

      try {
        const userDao = await import("../user.dao");
        await userDao.create(testUserInit);
      } catch (error) {
        err = error;
      }

      expect((err as EnvironmentConfigError).errorCode).toBe(
        ErrorCode.ENVIRONMENT_CONFIG_ERROR
      );
    });
  });

  describe("sad path - username unavailable", () => {
    let err: unknown;

    beforeAll(async () => {
      jest.resetModules();
      jest.doMock("/opt/nodejs/dynamo.config", () => ({
        dynamoClient: {
          put: jest.fn().mockReturnValueOnce({
            promise: jest
              .fn()
              .mockRejectedValueOnce(conditionalCheckFailedException),
          }),
        },
        TABLE_NAME: "test-table-name",
      }));
      jest
        .spyOn(console, "error")
        .mockClear()
        .mockImplementationOnce(jest.fn());

      try {
        const userDao = await import("../user.dao");
        await userDao.create(testUserInit);
      } catch (error) {
        err = error;
      }
    });

    it("should throw an error", async () => {
      expect(err).toMatchSnapshot();
    });

    it("should log error", () => {
      expect(jest.spyOn(console, "error").mock.calls).toMatchSnapshot();
    });
  });

  describe("sad path - DynamoDB put unknown error", () => {
    it("should throw an error", async () => {
      let err: unknown;

      jest.resetModules();
      jest.doMock("/opt/nodejs/dynamo.config", () => ({
        dynamoClient: {
          put: jest.fn().mockReturnValueOnce({
            promise: jest.fn().mockRejectedValueOnce(new Error("error")),
          }),
        },
        TABLE_NAME: "test-table-name",
      }));
      jest
        .spyOn(console, "error")
        .mockClear()
        .mockImplementationOnce(jest.fn());

      try {
        const userDao = await import("../user.dao");
        await userDao.create(testUserInit);
      } catch (error) {
        err = error;
      }

      expect(err).toMatchSnapshot();
    });
  });
});

describe("deleteUser", () => {
  describe("happy path", () => {
    it("should call DynamoDB delete", async () => {
      jest.resetModules();
      jest.doMock("/opt/nodejs/dynamo.config", () => ({
        dynamoClient: {
          delete: jest.fn().mockReturnValueOnce({
            promise: jest.fn().mockResolvedValueOnce(null),
          }),
        },
        TABLE_NAME: "test-table-name",
      }));
      const dynamoConfig = await import("/opt/nodejs/dynamo.config");
      const userDao = await import("../user.dao");
      await userDao.deleteUser(testUserInit.id);
      expect(
        (dynamoConfig.dynamoClient.delete as jest.Mock).mock.calls
      ).toMatchSnapshot();
    });
  });

  describe("sad path - DynamoDB table undefined", () => {
    it("should throw a EnvironmentConfigError error", async () => {
      let err: unknown;

      jest.resetModules();
      jest.doMock("/opt/nodejs/dynamo.config", () => ({
        TABLE_NAME: undefined,
      }));
      jest
        .spyOn(console, "error")
        .mockClear()
        .mockImplementationOnce(jest.fn());

      try {
        const userDao = await import("../user.dao");
        await userDao.deleteUser(testUserInit.id);
      } catch (error) {
        err = error;
      }

      expect((err as EnvironmentConfigError).errorCode).toBe(
        ErrorCode.ENVIRONMENT_CONFIG_ERROR
      );
    });
  });

  describe("sad path - user not found", () => {
    let err: unknown;

    beforeAll(async () => {
      jest.resetModules();
      jest.doMock("/opt/nodejs/dynamo.config", () => ({
        dynamoClient: {
          delete: jest.fn().mockReturnValueOnce({
            promise: jest
              .fn()
              .mockRejectedValueOnce(conditionalCheckFailedException),
          }),
        },
        TABLE_NAME: "test-table-name",
      }));
      jest
        .spyOn(console, "error")
        .mockClear()
        .mockImplementationOnce(jest.fn());
      try {
        const userDao = await import("../user.dao");
        await userDao.deleteUser(testUserInit.id);
      } catch (error) {
        err = error;
      }
    });

    it("should throw a UserNotFoundError error", () => {
      expect((err as UserNotFoundError).errorCode).toBe(
        ErrorCode.USER_NOT_FOUND
      );
    });

    it("should log error", () => {
      expect(jest.spyOn(console, "error").mock.calls).toMatchSnapshot();
    });
  });

  describe("sad path - DynamoDB delete unknown error", () => {
    it("should throw an error", async () => {
      let err: unknown;

      jest.resetModules();
      jest.doMock("/opt/nodejs/dynamo.config", () => ({
        dynamoClient: {
          delete: jest.fn().mockReturnValueOnce({
            promise: jest.fn().mockRejectedValueOnce(new Error("error")),
          }),
        },
        TABLE_NAME: "test-table-name",
      }));
      jest
        .spyOn(console, "error")
        .mockClear()
        .mockImplementationOnce(jest.fn());

      try {
        const userDao = await import("../user.dao");
        await userDao.deleteUser(testUserInit.id);
      } catch (error) {
        err = error;
      }
      expect(err).toMatchSnapshot();
    });
  });
});

describe("patch", () => {
  describe("happy path", () => {
    it("should call DynamoDB update", async () => {
      jest.resetModules();
      jest.doMock("/opt/nodejs/dynamo.config", () => ({
        dynamoClient: {
          update: jest.fn().mockReturnValueOnce({
            promise: jest.fn().mockResolvedValueOnce(null),
          }),
        },
        TABLE_NAME: "test-table-name",
      }));
      const dynamoConfig = await import("/opt/nodejs/dynamo.config");
      const userDao = await import("../user.dao");
      await userDao.patch(testUserInit);
      expect(
        (dynamoConfig.dynamoClient.update as jest.Mock).mock.calls
      ).toMatchSnapshot();
    });
  });

  describe("sad path - DynamoDB table undefined", () => {
    it("should throw a EnvironmentConfigError error", async () => {
      let err: unknown;

      jest.resetModules();
      jest.doMock("/opt/nodejs/dynamo.config", () => ({
        TABLE_NAME: undefined,
      }));
      jest
        .spyOn(console, "error")
        .mockClear()
        .mockImplementationOnce(jest.fn());

      try {
        const userDao = await import("../user.dao");
        await userDao.patch(testUserInit);
      } catch (error) {
        err = error;
      }

      expect((err as EnvironmentConfigError).errorCode).toBe(
        ErrorCode.ENVIRONMENT_CONFIG_ERROR
      );
    });
  });

  describe("sad path - missing ID", () => {
    it("should throw a RequiredPropertyMissingError error", async () => {
      let err: unknown;

      jest.resetModules();
      jest.doMock("/opt/nodejs/dynamo.config", () => ({
        TABLE_NAME: "test-table-name",
      }));
      jest
        .spyOn(console, "error")
        .mockClear()
        .mockImplementationOnce(jest.fn());

      try {
        const userDao = await import("../user.dao");
        await userDao.patch({ ...testUserInit, id: undefined });
      } catch (error) {
        err = error;
      }

      expect((err as RequiredPropertyMissingError).errorCode).toBe(
        ErrorCode.REQUIRED_PROPERTY_MISSING
      );
    });
  });

  describe("sad path - user not found", () => {
    let err: unknown;

    beforeAll(async () => {
      jest.resetModules();
      jest.doMock("/opt/nodejs/dynamo.config", () => ({
        dynamoClient: {
          update: jest.fn().mockReturnValueOnce({
            promise: jest
              .fn()
              .mockRejectedValueOnce(conditionalCheckFailedException),
          }),
        },
        TABLE_NAME: "test-table-name",
      }));
      jest
        .spyOn(console, "error")
        .mockClear()
        .mockImplementationOnce(jest.fn());
      const userDao = await import("../user.dao");
      try {
        await userDao.patch(testUserInit);
      } catch (error) {
        err = error;
      }
    });

    it("should throw a UserNotFoundError error", async () => {
      expect((err as UserNotFoundError).errorCode).toBe(
        ErrorCode.USER_NOT_FOUND
      );
    });

    it("should log error", () => {
      expect(jest.spyOn(console, "error").mock.calls).toMatchSnapshot();
    });
  });

  describe("sad path - DynamoDB patch unknown error", () => {
    it("should throw an error", async () => {
      let err: unknown;

      jest.resetModules();
      jest.doMock("/opt/nodejs/dynamo.config", () => ({
        dynamoClient: {
          update: jest.fn().mockReturnValueOnce({
            promise: jest.fn().mockRejectedValueOnce(new Error("error")),
          }),
        },
        TABLE_NAME: "test-table-name",
      }));
      jest
        .spyOn(console, "error")
        .mockClear()
        .mockImplementationOnce(jest.fn());

      try {
        const userDao = await import("../user.dao");
        await userDao.patch(testUserInit);
      } catch (error) {
        err = error;
      }
      expect(err).toMatchSnapshot();
    });
  });
});

describe("get", () => {
  describe("happy path", () => {
    it("should return the user", async () => {
      jest.resetModules();
      jest.doMock("/opt/nodejs/dynamo.config", () => ({
        dynamoClient: {
          get: jest.fn().mockReturnValueOnce({
            promise: jest.fn().mockResolvedValueOnce({
              Item: testUserInit,
            }),
          }),
        },
        TABLE_NAME: "test-table-name",
      }));
      const userDao = await import("../user.dao");
      expect(await userDao.get(testUserInit.id)).toMatchSnapshot();
    });
  });

  describe("sad path - DynamoDB table undefined", () => {
    it("should throw a EnvironmentConfigError error", async () => {
      let err: unknown;

      jest.resetModules();
      jest.doMock("/opt/nodejs/dynamo.config", () => ({
        TABLE_NAME: undefined,
      }));
      const userDao = await import("../user.dao");
      try {
        await userDao.get(testUserInit.id);
      } catch (error) {
        err = error;
      }

      expect((err as EnvironmentConfigError).errorCode).toBe(
        ErrorCode.ENVIRONMENT_CONFIG_ERROR
      );
    });
  });

  describe("sad path - DynamoDB get unknown error", () => {
    it("should throw an error", async () => {
      let err: unknown;

      jest.resetModules();
      jest.doMock("/opt/nodejs/dynamo.config", () => ({
        dynamoClient: {
          get: jest.fn().mockReturnValueOnce({
            promise: jest.fn().mockRejectedValueOnce(new Error("error")),
          }),
        },
        TABLE_NAME: "test-table-name",
      }));
      const userDao = await import("../user.dao");
      try {
        await userDao.get(testUserInit.id);
      } catch (error) {
        err = error;
      }
      expect(err).toMatchSnapshot();
    });
  });
});

describe("getByEmail", () => {
  describe("happy path", () => {
    it("should return the user", async () => {
      jest.resetModules();
      jest.doMock("/opt/nodejs/dynamo.config", () => ({
        dynamoClient: {
          query: jest.fn().mockReturnValueOnce({
            promise: jest.fn().mockResolvedValueOnce({
              Items: [testUserInit],
            }),
          }),
        },
        TABLE_NAME: "test-table-name",
        EMAIL_INDEX: "test-email-index",
      }));
      const userDao = await import("../user.dao");
      expect(await userDao.getByEmail(testUserInit.email)).toMatchSnapshot();
    });
  });

  describe("sad path - DynamoDB table undefined", () => {
    it("should throw a EnvironmentConfigError error", async () => {
      let err: unknown;

      jest.resetModules();
      jest.doMock("/opt/nodejs/dynamo.config", () => ({
        TABLE_NAME: undefined,
      }));
      const userDao = await import("../user.dao");
      try {
        await userDao.getByEmail(testUserInit.email);
      } catch (error) {
        err = error;
      }

      expect((err as EnvironmentConfigError).errorCode).toBe(
        ErrorCode.ENVIRONMENT_CONFIG_ERROR
      );
    });
  });

  describe("sad path - DynamoDB index undefined", () => {
    it("should throw a EnvironmentConfigError error", async () => {
      let err: unknown;

      jest.resetModules();
      jest.doMock("/opt/nodejs/dynamo.config", () => ({
        TABLE_NAME: "test-table-name",
        EMAIL_INDEX: undefined,
      }));
      const userDao = await import("../user.dao");
      try {
        await userDao.getByEmail(testUserInit.email);
      } catch (error) {
        err = error;
      }

      expect((err as EnvironmentConfigError).errorCode).toBe(
        ErrorCode.ENVIRONMENT_CONFIG_ERROR
      );
    });
  });

  describe("sad path - DynamoDB get unknown error", () => {
    it("should throw an error", async () => {
      let err: unknown;

      jest.resetModules();
      jest.doMock("/opt/nodejs/dynamo.config", () => ({
        dynamoClient: {
          query: jest.fn().mockReturnValueOnce({
            promise: jest.fn().mockRejectedValueOnce(new Error("error")),
          }),
        },
        TABLE_NAME: "test-table-name",
      }));
      const userDao = await import("../user.dao");
      try {
        await userDao.getByEmail(testUserInit.email);
      } catch (error) {
        err = error;
      }
      expect(err).toMatchSnapshot();
    });
  });
});
