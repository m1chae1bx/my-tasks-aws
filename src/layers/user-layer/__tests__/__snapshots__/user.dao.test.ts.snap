// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create sad path - DynamoDB put unknown error should throw an error 1`] = `[Error: error]`;

exports[`create sad path - username unavailable should log error 1`] = `
Array [
  Array [
    Object {
      "code": "ConditionalCheckFailedException",
      "message": "The conditional request failed",
      "name": "ConditionalCheckFailedException",
      "time": 2022-01-01T00:00:00.000Z,
    },
  ],
]
`;

exports[`create sad path - username unavailable should throw an error 1`] = `
UsernameUnavailableError {
  "errorCode": "usernameUnavailable",
  "message": "Username is already taken",
  "name": "UsernameUnavailableError",
}
`;

exports[`deleteUser happy path should call DynamoDB delete 1`] = `
Array [
  Array [
    Object {
      "ConditionExpression": "attribute_exists(PK)",
      "Key": Object {
        "PK": "USER#testUserId",
        "SK": "USER#testUserId",
      },
      "TableName": "test-table-name",
    },
  ],
]
`;

exports[`deleteUser sad path - DynamoDB delete unknown error should throw an error 1`] = `[Error: error]`;

exports[`deleteUser sad path - user not found should log error 1`] = `
Array [
  Array [
    Object {
      "code": "ConditionalCheckFailedException",
      "message": "The conditional request failed",
      "name": "ConditionalCheckFailedException",
      "time": 2022-01-01T00:00:00.000Z,
    },
  ],
]
`;

exports[`get happy path should return the user 1`] = `
Object {
  "email": "test@email.com",
  "fullName": "Full Name",
  "hash": "test-hash",
  "id": "testUserId",
  "nickname": "Nickname",
  "preferences": Object {
    "defaultListId": "test-default-list-id",
  },
  "salt": "test-salt",
}
`;

exports[`get sad path - DynamoDB get unknown error should throw an error 1`] = `[Error: error]`;

exports[`getByEmail happy path should return the user 1`] = `
Object {
  "email": "test@email.com",
  "fullName": "Full Name",
  "hash": "test-hash",
  "id": "testUserId",
  "nickname": "Nickname",
  "preferences": Object {
    "defaultListId": "test-default-list-id",
  },
  "salt": "test-salt",
}
`;

exports[`getByEmail sad path - DynamoDB get unknown error should throw an error 1`] = `
EnvironmentConfigError {
  "errorCode": "environmentConfigError",
  "message": "Environment variable EMAIL_INDEX is not set",
  "name": "EnvironmentConfigError",
}
`;

exports[`patch happy path should call DynamoDB update 1`] = `
Array [
  Array [
    Object {
      "ConditionExpression": "attribute_exists(PK)",
      "ExpressionAttributeValues": Object {
        ":d": "test-default-list-id",
        ":f": "Full Name",
        ":n": "Nickname",
      },
      "Key": Object {
        "PK": "USER#testUserId",
        "SK": "USER#testUserId",
      },
      "TableName": "test-table-name",
      "UpdateExpression": "set fullName = :f, nickname = :n, preferences.defaultListId = :d",
    },
  ],
]
`;

exports[`patch sad path - DynamoDB patch unknown error should throw an error 1`] = `[Error: error]`;

exports[`patch sad path - user not found should log error 1`] = `
Array [
  Array [
    Object {
      "code": "ConditionalCheckFailedException",
      "message": "The conditional request failed",
      "name": "ConditionalCheckFailedException",
      "time": 2022-01-01T00:00:00.000Z,
    },
  ],
]
`;
